azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py

```diff
@@ -22,66 +28,43 @@
 
 logger = get_logger(__name__)
 
-resource_test_env_vars = [
-    "azext_dt_ep_eventhub_namespace",
-    "azext_dt_ep_eventhub_policy",
-    "azext_dt_ep_eventhub_topic",
-    "azext_dt_ep_servicebus_namespace",
-    "azext_dt_ep_servicebus_policy",
-    "azext_dt_ep_servicebus_topic",
-    "azext_dt_ep_eventgrid_topic",
-    "azext_dt_ep_rg",
-]
-
-settings = DynamoSettings(opt_env_set=resource_test_env_vars)
-run_resource_tests = False
-run_endpoint_route_tests = False
-
-
-if all(
-    [
-        settings.env.azext_dt_ep_eventhub_namespace,
-        settings.env.azext_dt_ep_eventhub_policy,
-        settings.env.azext_dt_ep_eventhub_topic,
-        settings.env.azext_dt_ep_servicebus_namespace,
-        settings.env.azext_dt_ep_servicebus_policy,
-        settings.env.azext_dt_ep_servicebus_topic,
-        settings.env.azext_dt_ep_eventgrid_topic,
-        settings.env.azext_dt_ep_rg,
-    ]
-):
-    run_endpoint_route_tests = True
-
 
 class TestDTResourceLifecycle(DTLiveScenarioTest):
     def __init__(self, test_case):
         super(TestDTResourceLifecycle, self).__init__(test_case)
+        self.ensure_eventgrid_resource()
+        self.ensure_eventhub_resource()
+        self.ensure_servicebus_resource()
+
+    def tearDown(self):
+        super().tearDown()
+        try:
+            self.delete_eventhub_resources()
+        except Exception:
+            logger.info("The EventHub resources have already been deleted.")
```

Comment:
nit, there could potentially be some other reasons for the exception, but we should get additional logs from the embedded CLI call to help with diagnosing.

Reply:
Added better logging